yazi-core/src/manager/commands/bulk_rename.rs

```diff
@@ -117,4 +117,90 @@ impl Manager {
 		stdin().read_exact(&mut [0]).await?;
 		Ok(())
 	}
+
+	fn prioritized_paths(old: Vec<PathBuf>, new: Vec<PathBuf>) -> Vec<(PathBuf, PathBuf)> {
+		let orders: HashMap<_, _> = old.iter().enumerate().map(|(i, p)| (p, i)).collect();
+		let mut incomes: HashMap<_, _> = old.iter().map(|p| (p, false)).collect();
+		let mut todos: HashMap<_, _> = old
+			.iter()
+			.zip(new)
+			.map(|(o, n)| {
+				incomes.get_mut(&n).map(|b| *b = true);
+				(o, n)
+			})
+			.collect();
+
+		let mut sorted = Vec::with_capacity(old.len());
+		while !todos.is_empty() {
+			// Paths that are non-incomes and don't need to be prioritized in this round
+			let mut outcomes: Vec<_> = incomes.iter().filter(|(_, &b)| !b).map(|(&p, _)| p).collect();
+			outcomes.sort_unstable_by(|a, b| orders[b].cmp(&orders[a]));
```

Comment:
Is there any reason to use `sort_unstable_by` instead of `sort_by`? for in-place?

Reply:
Yeah, stable needs n/2 auxiliary memory to sort n elements in worst case